references/oidc-mock/apiproxy/policies/RF-RedirectURI-Control.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!--
+ Copyright 2020 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+<RaiseFault async="false" continueOnError="false" enabled="true" name="RF-RedirectURI-Control">
+    <DisplayName>RF-RedirectURI-Control</DisplayName>
+    <Properties/>
+    <FaultResponse>
+        <Set>
+            <Headers>
+                <Header name="Cache-Control">no-store</Header>
+                <Header name="Pragma">no-cache</Header>
+            </Headers>
+            <Payload contentType="application/json">{"error":"invalid_request"}</Payload>
```

Comment:
In the cases where the request is triggered by a Web Client (e.g. `GET /authorize` `GET /consent`) we should return a very simple HTML page with the error instead of JSON if we don't have a valid client ID and redirect URI.

If we do have a valid client ID and redirect URI, we should send the error via 302 redirect (url)?error=xxx